SRC/FileOperation.cpp

```diff
@@ -121,58 +123,69 @@ int FileOperation::getLinesCount()
 
 void FileOperation::CreateDB_Folder()
 {
-	string cmd = "mkdir " + database_folder;
-	system(cmd.c_str());
-	cmd = "mkdir " + database_folder + "/" + file_name + "_db";
-	system(cmd.c_str());
+	string cmd = CMDMKDIR + string(SPACE) + database_folder + SEMICOLON + CMDMKDIR + string(SPACE) + database_folder + file_name + DB_EXT;
+#ifdef _WIN32
+	FILE* pipe = _popen(cmd.c_str(), "r");
+#else
+	FILE* pipe = popen(cmd.c_str(), "r");
+#endif
+	if (!pipe) {
+		cout << "popen failed! for CreateDB_Folder" << endl;
+	}
 
-	database_folder = database_folder + "/" + file_name + "_db";
+#ifdef _WIN32
+	_pclose(pipe);
+#else
+	pclose(pipe);
+#endif
+	database_folder = database_folder + file_name + DB_EXT;
 	cout << "\n Called CreateDB_Folder " << database_folder;
 }
 
 void FileOperation::copyOriginalFile()
 {
-	string folder = database_folder + "/original";
-	string cmd = "mkdir " + folder;
-	system(cmd.c_str());
-	cmd = "cp " + file_path + "/" + file_name +" " + folder;
-	system(cmd.c_str());
+	string folder = database_folder + SEPERATOR + ORIGINAL;
+	string cmd = CMDMKDIR + string(SPACE) + folder + SEMICOLON + CMDCP + string(SPACE) + file_path + SEPERATOR + file_name + SPACE + folder;
+
+#ifdef _WIN32
+	FILE* pipe = _popen(cmd.c_str(), "r");
+#else
+	FILE* pipe = popen(cmd.c_str(), "r");
+#endif
+	if (!pipe) {
+		cout << "popen failed! for copyOriginalFile" << endl;
+	}
+#ifdef _WIN32
+	_pclose(pipe);
+#else
+	pclose(pipe);
+#endif
 	original_file_folder = folder;
-	cout << "\n Called copyOriginalFile " << original_file_folder;
+	//cout << "\n Called copyOriginalFile " << original_file_folder;
 }
 
 
-void FileOperation::ReplaceOriginalFile()
+void FileOperation::ReplaceOriginalFile() const

```

Comment:
Can we check if we can delete the file after replacing the original file.

Reply:
Changes raised in new PR.